<launch>
    <arg name="topic_group" default="arm_controller"/>
    <arg name="rname" default="z1"/>
    <arg name="z1robot_path" value="(find $(arg rname)_description)"/>
    <arg name="z1robot_config_path" value="(find $(arg rname)_moveit_config)"/>
    <arg name="dollar" value="$"/>
    <arg name="UnitreeGripperYN" default="false"/>
    <param name="z1robot_description" command="$(find xacro)/xacro --inorder '$(arg dollar)$(arg z1robot_path)/xacro/robot.xacro' UnitreeGripper:=$(arg UnitreeGripperYN)"/>

    <node pkg="arm_controller" type="arm_controller_node" name="arm_controller_node" output="screen">
        <remap from="/joint_states" to="/$(arg topic_group)/joint_states"/>
        <remap from="/cmd_joint_states" to="/$(arg topic_group)/cmd_joint_states" />
        <remap from="/end_effector_pose" to="/$(arg topic_group)/end_effector_pose"/>
        <remap from="/execute_process" to="/$(arg topic_group)/execute_process"/>
        <remap from="/plan" to="/$(arg topic_group)/plan"/>
        <remap from="/back_to_home" to="/$(arg topic_group)/back_to_home"/>
        <remap from="/check_pose_in_workspace" to="/$(arg topic_group)/check_pose_in_workspace"/>
    </node>

    <!-- <node pkg="robot_state_publisher" type="robot_state_publisher" name="arm_state_publisher">
        <remap from="/$(arg topic_group)/joint_states" to="/$(arg topic_group)_rviz/joint_states" />
    </node> -->
</launch>
